config/v3_0/types/config.go

```diff
@@ -24,3 +29,34 @@ var (
 		Minor: 0,
 	}
 )
+
+func (cfg Config) Validate(c path.ContextPath) (r report.Report) {
+	systemdPath := "/etc/systemd/system/"
+	unitPaths := map[string]struct{}{}
+	for _, unit := range cfg.Systemd.Units {
+		if !util.NilOrEmpty(unit.Contents) {
+			pathString := systemdPath + unit.Name
+			unitPaths[pathString] = struct{}{}
+		}
+		for _, dropin := range unit.Dropins {
+			pathString := systemdPath + unit.Name + ".d/" + dropin.Name
+			unitPaths[pathString] = struct{}{}
```

Comment:
Looks like we lost the `Contents` check for dropins.

Reply:
> Unfortunately, if dropin `contents` is the empty string, we write the file, but if unit `contents` is the empty string, we don't. 

I misunderstood this to mean that `Contents` did not need to be specified for dropins for the file to be written.